ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -200,6 +200,15 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	rayDashboardClient := utils.GetRayDashboardClientFunc()
 	rayDashboardClient.InitClient(clientURL)
 
+	// Check the dashboard readiness by checking the err return from rayDashboardClient.GetJobInfo.
```

Comment:
We do not need to send a request to check the dashboard's readiness in each reconciliation. We can only check it when no Kubernetes Job exists.

https://github.com/ray-project/kuberay/blob/6d5020fba579ca3296a20fc8ee6a6b809864690d/ray-operator/controllers/ray/rayjob_controller.go#L331

Reply:
I think line 331 is too far from the dashboard related code. Inserting readiness check there might be a little confusing. 

What about checking the readiness right before setting the `rayJobInstance.Status.DashboardURL`? And we don't check it again if `rayJobInstance.Status.DashboardURL` is set.